src/compile/mark/mark.ts

```diff
@@ -36,7 +38,7 @@ export function parseMark(model: UnitModel): any[] {
 function parsePathMark(model: UnitModel) { // TODO: extract this into compilePathMark
   const mark = model.mark();
   // TODO: replace this with more general case for composition
-  const hasParentData = !!model.parent();
+  const isFaceted = model.parent() && model.parent().isFacet();
```

Comment:
Ok.  I think you're right about `layer` with children deriving data from parents because we don't add additional group.

However, this still seems wrong if the facet is a grandparent:

```
{
   data: ...,
   facet: ...,
   spec: {
    layer:  [{  
        // no data here and its parent is not a facet. 
        mark: ...
        encoding: ...
    }, {  
        // no data here
        mark: ...
        encoding: ...
    }]
  }
}
```

and is also wrong in the case of splom.

The right name should be `useFacetData`  (which can be parent / grandparent in the model tree + only refer to the case where the facet children doesn't have its own data).